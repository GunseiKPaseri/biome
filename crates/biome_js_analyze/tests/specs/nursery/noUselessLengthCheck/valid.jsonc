[
  "if (array.every(Boolean));",
  "if (array.some(Boolean));",
  "const isAllTrulyOrEmpty = array.every(Boolean);",
  "if (array.length === 0 || anotherCheck() || array.every(Boolean));",
  "const isNonEmptyAllTrulyArray = array.length > 0 && array.every(Boolean);",
  "const isEmptyArrayOrAllTruly = array.length === 0 || array.some(Boolean);",
  "array.length === 0 ?? array.every(Boolean)",
  "array.length === 0 && array.every(Boolean)",
  "(array.length === 0) + (array.every(Boolean))",
  "array.length === 1 || array.every(Boolean)",
  "array.length === '0' || array.every(Boolean)",
  "array.length === 0. || array.every(Boolean)",
  "array.length === 0x0 || array.every(Boolean)",
  "array.length !== 0 || array.every(Boolean)",
  "array.length == 0 || array.every(Boolean)",
  "0 === array.length || array.every(Boolean)",
  "array?.length === 0 || array.every(Boolean)",
  "array.notLength === 0 || array.every(Boolean)",
  "array[length] === 0 || array.every(Boolean)",
  "array.length === 0 || array.every?.(Boolean)",
  "array.length === 0 || array?.every(Boolean)",
  "array.length === 0 || array.every",
  "array.length === 0 || array[every](Boolean)",
  "array1.length === 0 || array2.every(Boolean)",
  "(foo && array.length === 0) || array.every(Boolean) && foo",
  "array.length === 0 || (array.every(Boolean) && foo)",
  "(foo || array.length > 0) && array.some(Boolean)",
  "array.length > 0 && (array.some(Boolean) || foo)",
  "array.length > 0 && array.some(Boolean) && array.other(Boolean)",
  "array.length > 0 && array.some(Boolean) && array.every(Boolean)",
  "array.length === 0",
  "array.every(Boolean)"
]