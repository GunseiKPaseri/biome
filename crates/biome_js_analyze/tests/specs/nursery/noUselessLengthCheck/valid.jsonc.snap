---
source: crates/biome_js_analyze/tests/spec_tests.rs
assertion_line: 86
expression: valid.jsonc
---
# Input
```cjs
if (array.every(Boolean));
```

# Input
```cjs
if (array.some(Boolean));
```

# Input
```cjs
array.length === 5 && array.some(Boolean)
```

# Input
```cjs
const isAllTrulyOrEmpty = array.every(Boolean);
```

# Input
```cjs
if (array.length === 0 || anotherCheck() || array.every(Boolean));
```

# Input
```cjs
const isNonEmptyAllTrulyArray = array.length > 0 && array.every(Boolean);
```

# Input
```cjs
const isEmptyArrayOrAllTruly = array.length === 0 || array.some(Boolean);
```

# Input
```cjs
array.length === 0 ?? array.every(Boolean)
```

# Input
```cjs
array.length === 0 && array.every(Boolean)
```

# Input
```cjs
(array.length === 0) + (array.every(Boolean))
```

# Input
```cjs
array.length === 1 || array.every(Boolean)
```

# Input
```cjs
array.length === '0' || array.every(Boolean)
```

# Input
```cjs
array.length === 0. || array.every(Boolean)
```

# Input
```cjs
array.length === 0x0 || array.every(Boolean)
```

# Input
```cjs
array.length !== 0 || array.every(Boolean)
```

# Input
```cjs
array.length == 0 || array.every(Boolean)
```

# Input
```cjs
0 === array.length || array.every(Boolean)
```

# Input
```cjs
array?.length === 0 || array.every(Boolean)
```

# Input
```cjs
array.notLength === 0 || array.every(Boolean)
```

# Input
```cjs
array[length] === 0 || array.every(Boolean)
```

# Input
```cjs
array.length === 0 || array.every?.(Boolean)
```

# Input
```cjs
array.length === 0 || array?.every(Boolean)
```

# Input
```cjs
array.length === 0 || array.every
```

# Input
```cjs
array.length === 0 || array[every](Boolean)
```

# Input
```cjs
array1.length === 0 || array2.every(Boolean)
```

# Input
```cjs
(foo && array.length === 0) || array.every(Boolean) && foo
```

# Input
```cjs
array.length === 0 || (array.every(Boolean) && foo)
```

# Input
```cjs
(foo || array.length > 0) && array.some(Boolean)
```

# Input
```cjs
array.length > 0 && (array.some(Boolean) || foo)
```

# Input
```cjs
array.length > 0 && array.some(Boolean) && array.other(Boolean)
```

# Input
```cjs
array.length > 0 && array.some(Boolean) && array.every(Boolean)
```

# Input
```cjs
array.length === 0
```

# Input
```cjs
array.every(Boolean)
```
